benchmark/Cpp/Savina/src/concurrency/SleepingBarber.lf

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright (C) 2020 TU Dresden
+ *
+ * In this benchmark, there are four main components: The barber, a waiting
+ * room, a customer factory and the customers. The customer factory controls the
+ * production of new customers and sends them to the waiting room. If the
+ * waiting room is full, then the customers return and try again later. If the
+ * waiting room has empty seats, then the customer enters and waits to be served
+ * by the barber. In the Savina implementation, the waiting room takes the role
+ * of a central manager. If the room is empty, it tells the barber to go sleep.
+ * If a customer comes and the barber sleeps, the waiting room wakes up the
+ * barber and sends the customer in to the barber to be served. When the barber
+ * finishes, the waiting room sends in the next customer. If there are no more
+ * customers, the waiting room tells the barber to sleep again.
+ *
+ * In the Akka implementation production of customers and processing at the
+ * barber happens at certain rates. The workload of the factory and the barber
+ * has a random length and the average workload duration is controlled by the
+ * rate parameters. Depending on the rates selected and the precise runtime
+ * scheduling, the waiting room will have a constantly changing state in an
+ * actor implementation.
+ *
+ * In an LF implementation, the processing time of the workload does not
+ * influence the logical arrival and processing rates. Even with a random
+ * workload, the customer factory and the barber could operate logically
+ * synchronously in LF. To mimic the behaviour of the original Akka
+ * implementation and to have a more realistic setup, this LF implementation
+ * also uses varying logical delays. The factory will produce customers with a
```

Comment:
Why not use physical actions?

Reply:
Good point. I changed that and it even performs faster. I think the problem with physical actions here is that they force sequential operation. It's not a large problem in this case though, since the workload of the factory and the barber is barely noticeable.